test/functions.js

```diff
@@ -694,6 +694,9 @@
       assert.deepEqual(_.toArray(cb(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)), _.range(1, 11));
     });
 
+    var deepProperty = _.iteratee(['a', 'b']);
+    assert.equal(deepProperty({a: {b: 2}}), 2, 'treats an array as a deep property accessor');
+
```

Comment:
@jdalton Thanks for bringing this up. I'm a bit confused though. In lodash you use arrays as shorthand for a type of predicate which can already be expressed via an object (`_.matches`)?

I'm guessing from the phrase "fast path" and the perf graph you included that this somehow offers a performance gain? (Or maybe I'm misunderstanding?)

To me, deep property matching seems much more useful than a performance increase which users could presumably get in the places where performance was a concern by simply using `_.matchesProperty()` or their own optimized function directly.

Again, I could just be misinterpreting your comment, so feel free to set me straight. :smile: